c/secp256k1_blake160_sighash_all.c

```diff
@@ -24,48 +24,65 @@
 #define RECID_INDEX 64
 /* 32 KB */
 #define WITNESS_SIZE 32768
+#define SCRIPT_SIZE 32768
+#define SIGNATURE_SIZE 65
 
 /*
  * Arguments are listed in the following order:
- * 0. program name
- * 1 ~ n. pubkey blake160 hash, blake2b hash of pubkey first 20 bytes, used to
+ * 0 ~ n. pubkey blake160 hash, blake2b hash of pubkey first 20 bytes, used to
  * shield the real pubkey in lock script, the lock is considered as a multisig
  * lock when the number of pubkeys is more than 1.
  * n + 1. multisig threshold (optional), used to indicate the threshold of a
  * multisig lock, must ignore this arg when only one pubkey is used.
  *
  * Witness:
  * 0 ~ m signature, signatures used to present ownership, the number of signatures
- * must be equals to the threshold if the lock is a multisig lock, otherwise must 
+ * must be equals to the threshold if the lock is a multisig lock, otherwise must
  * use only 1 signature.
  */
-int main(int argc, char* argv[]) {
+int main() {
   int ret;
-  int recid;
   size_t index = 0;
   uint64_t sigs_cnt = 0;
   uint64_t threshold = 0;
   volatile uint64_t len = 0;
   unsigned char tx_hash[BLAKE2B_BLOCK_SIZE];
   unsigned char temp[TEMP_SIZE];
   unsigned char witness[WITNESS_SIZE];
+  unsigned char script[SCRIPT_SIZE];
   uint8_t secp_data[CKB_SECP256K1_DATA_SIZE];
-  mol_pos_t witness_pos;
-  size_t witness_len;
-  mol_read_res_t arg_res;
+  mol_pos_t script_pos;
+  mol_read_res_t args_res;
   mol_read_res_t bytes_res;
 
-  if (argc < 2 || argc == 3) {
+  /* Load args */
+  len = SCRIPT_SIZE;
+  ret = ckb_load_script(script, &len, 0);
+  if (ret != CKB_SUCCESS) {
+    return ERROR_SYSCALL;
+  }
+  script_pos.ptr = (const uint8_t*)script;
+  script_pos.size = len;
+  args_res = mol_cut(&script_pos, MOL_Script_args());
+  if (args_res.code != 0) {
+    return ERROR_ENCODING;
+  }
+  bytes_res = mol_cut_bytes(&args_res.pos);
+  if (bytes_res.code != 0) {
+    return ERROR_ENCODING;
+  } else if (bytes_res.pos.size == 0) {
     return ERROR_WRONG_NUMBER_OF_ARGUMENTS;
-  } else if (argc == 2) {
-    sigs_cnt = 1;
-    threshold = 1;
+  }
+
+  /* Calculate sigs count and threshold */
+  sigs_cnt = bytes_res.pos.size / BLAKE160_SIZE;
+  if (bytes_res.pos.size % BLAKE160_SIZE == 0) {
+    threshold = sigs_cnt;
   } else {
-    sigs_cnt = argc - 2;
-    threshold = argv[argc - 1][0];
+    threshold = bytes_res.pos.ptr[sigs_cnt * BLAKE160_SIZE];
   }
 
-  if (threshold < 1) {
```

Comment:
Miss this check that threshold cannot be zero.

Reply:
Updated, and added `test_invalid_threshold` unit test, please review again, thanks.